src/include/distributed/multi_router_executor.h

```diff
@@ -9,15 +9,37 @@
 #ifndef MULTI_ROUTER_EXECUTOR_H_
 #define MULTI_ROUTER_EXECUTOR_H_
 
+#include "c.h"
+
+#include "access/sdir.h"
 #include "distributed/multi_physical_planner.h"
 #include "executor/execdesc.h"
+#include "nodes/pg_list.h"
+
+/* maximum (textual) lengths of hostname and port */
+#define MAX_NODE_LENGTH 255
```

Comment:
How did we arrive at 255? IIRC that's roughly the length of the hostname limits, without the port itself. If we added 10 to that, we could handle all possible URL/port combinations.